src/TraceEvent/Symbols/SymbolReader.cs

```diff
@@ -2699,9 +2699,28 @@ unsafe internal SourceFile(SymbolModule module, IDiaSourceFile sourceFile)
             }
         }
 
-        private static byte[] ComputeHash(string filePath)
+        private byte[] ComputeHash(string filePath)
         {
-            System.Security.Cryptography.MD5CryptoServiceProvider crypto = new System.Security.Cryptography.MD5CryptoServiceProvider();
+            System.Security.Cryptography.HashAlgorithm crypto = null;
+            switch (m_hashType)
+            {
+                case 1:
+                    crypto = new System.Security.Cryptography.MD5CryptoServiceProvider();
+                    break;
+
+                case 2:
+                    crypto = new System.Security.Cryptography.SHA1CryptoServiceProvider();
+                    break;
+
+                case 3: 
+                    crypto = new System.Security.Cryptography.SHA256CryptoServiceProvider();
+                    break;
+
+                default:
+                    Debug.Assert(false);
```

Comment:
Don't just assert, throw an IllegalOperationException.   Otherwise we will nullRef below which is unhelpful.